src/main/java/frc/robot/subsystems/scoring/ScoringSubsystem.java

```diff
@@ -0,0 +1,126 @@
+package frc.robot.subsystems.scoring;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.subsystems.shooter.ShooterIOInputsAutoLogged;
+
+public class ScoringSubsystem extends SubsystemBase {
+    private final ShooterIO shooterIo;
+    private final ShooterIOInputsAutoLogged shooterInputs = new ShooterIOInputsAutoLogged();
+
+    private final AimerIO aimerIo;
+    private final AimerIOInputsAutoLogged aimerInputs = new AimerIOInputsAutoLogged();
+
+    private final Timer timer = new Timer();
+
+    private enum ScoringState {
+        WAIT,
+        INTAKE,
+        PRIME,
+        SHOOT,
+        ENDGAME
+    }
+
+    private ScoringState state = ScoringState.WAIT;
+
+    public ScoringSubsystem(ShooterIO shooterIo, AimerIO aimerIo) {
+        this.shooterIo = shooterIo;
+        this.aimerIo = aimerIo;
+    }
+
+    public void intake() {
+        if (state == ScoringState.WAIT) {
+            enterIntake();
+        }
+    }
+
+    public void aim() {
+        if (state == ScoringState.WAIT && true) { // TODO: Add banner sensor
+            enterPrime();
+        }
+    }
+
+    public void shoot() {
+        if (state == ScoringState.PRIME) {
+            enterShoot();
+        }
+    }
+
+    public void abort() {
+        if (state == ScoringState.PRIME || state == ScoringState.SHOOT) {
+            enterWait();
+        }
+    }
+
+    public void endgame() {
+        if (state == ScoringState.WAIT) {
+            enterEndgame();
+        }
+    }
+
+    private void enterWait() {
+        state = ScoringState.WAIT;
+
+        aimerIo.setAimAngRad(0);
+        shooterIo.setShooterVelRPM(0);
+        shooterIo.setKickerVolts(0);
+    }
+
+    private void enterIntake() {
+        state = ScoringState.INTAKE;
+
+        aimerIo.setAimAngRad(0);
+        shooterIo.setShooterVelRPM(-10);
+        shooterIo.setKickerVolts(-1);
+    }
+
+    private void enterPrime() {
+        state = ScoringState.PRIME;
+
+        shooterIo.setShooterVelRPM(100);
+    }
+
+    private void enterShoot() {
+        state = ScoringState.SHOOT;
+
+        shooterIo.setKickerVolts(1);
+        timer.reset();
+        timer.start();
+    }
+
+    private void enterEndgame() {
+        state = ScoringState.ENDGAME;
+    }
```

Comment:
As written, I'm not 100% sure what these do. But it seems like this code would be migrated to the state functions and then we could create a single function called `setAction` that sets the action we want to take (which will in turn exercise the state machine edges for us)